### ISSUE TITLE
Regression task gives nan output

### ISSUE THREAD

**Issue by aloreggia** [Lasagne/Lasagne#333]
I'm trying to run a regression task using Lasagne.
I set the network to output a softmax layer using a categorical cross entropy.
output_nonlinearity =  softmax,
objective_loss_function = categorical_crossentropy,

The network train successful for more or less 200/300 epochs and then it start giving nan values, take a look to an extract of the log:
    241       1.62136       3.61951      0.44795  1.04s
    242       1.62833       3.60246      0.45201  1.04s
    243       1.63284       3.57754      0.45641  1.04s
    244       1.63092           nan          nan  1.04s
    245           nan           nan          nan  1.04s
    246           nan           nan          nan  1.04s

Any idea why this is happening? 

Thanks
Andrea

**Comment by f0k** [comment]
> Any idea why this is happening?

Just to state the obvious: If the network predictions reach 0 or 1, you will definitely get NaNs. It's possible to get undefined values earlier than that for large inputs to the softmax function, though. You can try to use a log-domain softmax and corresponding cross-entropy loss instead: https://github.com/Lasagne/Lasagne/issues/332#issuecomment-122328992
If this fixes the problem, please do report back! We may include it into Lasagne then, or try to get a suitable graph optimizer into Theano.

**Comment by aloreggia** [comment]
I'm running several task with the your suggestions as in #332 and now everything seems to work quite well.

Thank you
